components/StarSearch/ShareChatMenu.tsx

```diff
@@ -44,13 +53,28 @@ export function ShareChatMenu({ shareUrl, copyLinkHandler, createLink }: ShareMe
         {createLink ? (
           <DropdownMenuItem className="pl-4 rounded-md">
             <button
-              className="flex items-center w-full gap-1"
+              className="flex items-center w-full gap-2"
               onClick={() => {
+                if (isCreatingLink) {
+                  return;
+                }
+
                 createLink();
+                setIsCreatingLink(true);
               }}
+              aria-disabled={isCreatingLink}
             >
-              <LinkIcon size={12} />
-              <span>Create Share Link</span>
+              {isCreatingLink ? (
+                <>
+                  <Spinner className="w-3 h-3 stroke-light-slate-12" />
```

Comment:
The shared `<Button />` component with the loading state set is not being used because all the styling of the button would have to be stripped to look good in the share menu.